### ISSUE TITLE
Compiler and eslint errors only reported if file has a used exports

### ISSUE THREAD

**Issue by jonasgheer** [TypeStrong/fork-ts-checker-webpack-plugin#479]
I currently have an existing javascript project and I'm looking for ways forward regarding a migration to typescript.

My setup is pretty close to the standard setup under "Installation":

```javascript
 module: {
    rules: [
      {
        test: /\.(jsx?|tsx?)$/,
        exclude: /node_modules/,
        use: [{loader: 'babel-loader', options: {...}}, {loader: 'ts-loader', options: {transpileOnly: true}}] 
      }
    ]
  },
  plugins: [new ForkTsCheckerWebpackPlugin({eslint: {files: "./app/js/**/*.{ts,tsx,js,jsx}"}})]
```

My tsconfig:
```json
{
  "compilerOptions": {
    "outDir": "./tsBuild",
    "jsx": "preserve",
    "target": "ES2020",
    "strict": true,
    "lib": ["dom", "ES2020"],
    "allowJs": true,
    "importsNotUsedAsValues": "preserve",
    "moduleResolution": "Node"
  },
  "include": [
     "./app/js/**/*"
  ]
}
```

I add a ts test file to an existing folder in my project. Lets say  `test.ts`

I add some invalid typescript:
```typescript
export function foo(n: cookie): boolean {
  return true;
}
```

I run the build using `webpack --mode development --watch`

I expect an compilation error from typescript, but I get "No issues found". Intellij has successfully pointed out the error.

I now import `foo` in a random react component and call it. All of a sudden, webpack reports the error in my console:
```
ERROR in app/js/util/test.ts 30:24-30
TS2304: Cannot find name 'cookie'.
    28 | 
    29 | 
  > 30 | export function foo(n: cookie): boolean {
       |                        ^^^^^^
    31 |     return true;
    32 | }
    33 | 

```

The same is true for eslint warnings and errors. Only if I export something from the file and use it somewhere does the file get checked.

**Comment by dmitrysteblyuk** [comment]
Why was this closed?

**Comment by piotr-oles** [comment]
I think it was closed because its a correct behaviour - webpack mode will report only files that are used in the webpack's compilation :)

**Comment by jonasgheer** [comment]
> I think it was closed because its a correct behaviour - webpack mode will report only files that are used in the webpack's compilation :)

You are correct. Woeful ignorance on my part.
